atomisticparsers/gromacs/parser.py

```diff
@@ -690,6 +689,27 @@ def parse_thermodynamic_data(self):
                     sec_energy.contributions.append(
                         EnergyEntry(kind=self._metainfo_mapping[key], value=val))
 
+        if not flag_system_time_map:
+            if n_evaluations == len(sec_system):
+                for i_calc, calc in enumerate(sec_run.calculation):
+                    calc.system_ref = sec_system[i_calc]
+                self.logger.warning('Time and step information not available, but system and calculation are the same length.'
+                                    'Assuming correspondence and creating references between these two lists.')
+                return
+
+        for time, n in self._system_time_map.items():
+            sec_scc = sec_run.m_create(Calculation)
+            sec_scc.time = time
+            sec_scc.step = int((time / time_step).magnitude)
+
+        times = [calc.time for calc in sec_run.calculation]
+        if None not in times:
```

Comment:
I do like this sorting, maybe we can get rid of this if we concatenate all times from calculation and system and loop over this.

Reply:
Definitely open to suggestions here, but not exactly sure what you mean.

`times` is the the concatenation of all times from calculation and system, already removing overlapping times...the calculation times come first and then the remaining system times are at the end. I am not sure how to avoid sorting here though, since the (non-)overlap of these two lists is completely arbitrary...?